<!-- 
  Launches the obstacle detection node
 -->
 
<launch>
  <arg name="downsample_leaf_size"        default="0.2"/>
  <arg name="input_topic"                 default="camera/depth/points"/>
  <arg name="filtered_obstacles_topic"    default="filtered_obstacles"/>
  <arg name="filtered_ground_plane_topic" default="filtered_ground_plane"/>
  
  <node pkg="obstacle_detection" type="lidar_image_detect.py" name="image_obstacle_detection" output="screen">
  </node>
  <node pkg="obstacle_detection" type="obstacle_detection" name="obstacle_detection" output="screen">
    <param name="downsample_leaf_size" type="double" value="$(arg downsample_leaf_size)"/>
    <param name="ground_min" value="-1.5"/>
    <param name="ground_max" value="1.0"/>
    <param name="cluster_tolerance" value="0.6"/>
    <param name="min_cluster_size" value="10"/>
    <param name="max_cluster_size" value="1500"/>
    <param name="distance_threshould" value="0.08"/>
    <remap from="input"                       to="$(arg input_topic)"/>
    <remap from="filtered_obstacles"          to="$(arg filtered_obstacles_topic)"/>
    <remap from="filtered_ground_plane_topic" to="$(arg filtered_ground_plane_topic)"/>
  </node>
</launch>
